src/test/java/net/kaczmarzyk/NotEmptyE2eTest.java

```diff
@@ -0,0 +1,176 @@
+
+    @Controller
+    @RequestMapping("/notEmpty")
+    private static class NotEmptySpecController {
+
+        @Autowired
+        CustomerRepository customerRepo;
+
+        @RequestMapping(value = "/customerOrders")
+        @ResponseBody
+        public Object findByNotEmptyOrders(
+                @Spec(path="orders", params="notEmptyOrders", spec=NotEmpty.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+
+        @RequestMapping(value = "/customerPhoneNumbers")
+        @ResponseBody
+        public Object findByNotEmptyPhoneNumbers(
+                @Spec(path="phoneNumbers", params="notEmptyPhoneNumbers", spec=NotEmpty.class) Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+
+        @RequestMapping(value = "/customersWithNotEmptyOrders_constVal")
+        @ResponseBody
+        public Object findByNotEmptyOrdersConstVal(
+                @Spec(path="orders", spec=NotEmpty.class, constVal = "true") Specification<Customer> spec) {
+
+            return customerRepo.findAll(spec);
+        }
+    }
+
+    @Before
+    public void initData() {
+        customer("Barry", "Benson")
+                .phoneNumbers("123456789")
+                .build(em);
+        customer("Vanessa", "Bloom")
+                .orders(order("flowers"))
+                .build(em);
+    }
+
+    @Test
+    public void findsByNotEmptyCustomerOrders_oneToManyAssociation() throws Exception {
+        mockMvc.perform(get("/notEmpty/customerOrders")
+                        .param("notEmptyOrders", "true")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].firstName").value("Homer"))
+                .andExpect(jsonPath("$[1].firstName").value("Bart"))
+                .andExpect(jsonPath("$[2].firstName").value("Moe"))
+                .andExpect(jsonPath("$[3].firstName").value("Ned"))
+                .andExpect(jsonPath("$[4].firstName").value("Vanessa"))
+                .andExpect(jsonPath("$[5]").doesNotExist());
+    }
+
+    @Test
+    public void findsByNotEmptyCustomerPhoneNumbers_elementCollectionWithSimpleNonEntityValues() throws Exception {
+        mockMvc.perform(get("/notEmpty/customerPhoneNumbers")
+                        .param("notEmptyPhoneNumbers", "true")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].firstName").value("Barry"))
+                .andExpect(jsonPath("$[1]").doesNotExist());
+    }
+
+    @Test
+    public void findsByEmptyCustomerOrders_oneToManyAssociation() throws Exception {
+        mockMvc.perform(get("/notEmpty/customerOrders")
+                        .param("notEmptyOrders", "false")
```

Comment:
please check (may be in a separate e2e test case) if it will work if we skip the parameter here, but use @RequestParam with "defaultValue" (will it be passed to SAR? If so, then great, keep this test. If not, then let's create an issue for that.